packages/react-apollo/src/provider.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useContext, useEffect, useState } from 'react'
+
+import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client'
+import { createApolloClient, NhostApolloClientOptions } from '@nhost/apollo'
+import { NhostReactContext } from '@nhost/react'
+
+// TODO check this type
+type Props = Omit<NhostApolloClientOptions, 'backendUrl' | 'authService'>
+const Wrapper: React.FC<Props> = ({ children, ...options }) => {
+  const { interpreter, backendUrl } = useContext(NhostReactContext)
+
+  const [client, setClient] = useState<ApolloClient<InMemoryCache>>()
+
+  useEffect(() => {
+    setClient(createApolloClient({ interpreter, backendUrl, ...options }))
```

Comment:
If we just simply want to create an Apollo client when the provider component is mounted it would be enough to create this client outside of the `useEffect` call. That would improve the readability of this component. But let's discuss it, probably I'm missing something that makes this hook necessary.

If the `useEffect` can be removed, then we know for sure that there is no such case when a `client` is not available, thus the `return <div>no Apollo client</div>` can be removed. If there is a case where `client` is not available, then it can stay here to provide a fallback.

I'd also convert it to a more readable format then with an early-exit / if-guard approach:

```jsx
if (client) {
  return <ApolloProvider ...>...</ApolloProvider>
}

return <div>no Apollo client</div>
```

Reply:
it was related to an attempt to make Apollo and Nextjs work together, but I set this aside (it's a real pain), so I simplified in the latest commit. Thanks!

Reply:
When using Next, if `createClient` is not wrapped into `useEffect`, then it seeks to render the page on the server side. This is to avoid as we don't want SSR if no `getServerSideProps` is present.
Plus, for the moment, Apollo is not correctly set with the sdk to run on the server side, then to hydrate the apollo cache.
Wrapping again in an `useEffect` until we found a better solution